lib/storage/data/external/AwsClient.js

```diff
@@ -168,50 +185,58 @@ class AwsClient {
                 logHelper(log, logLevel, 'error heading object ' +
                 'from datastore', err, this._dataStoreName);
                 return callback(retError);
-            }
-            return callback(null, data);
         });
     }
     get(objectGetInfo, range, reqUids, callback) {
         const log = createLogger(reqUids);
         const { key, dataStoreVersionId } = objectGetInfo;
-        const request = this._client.getObject({
+        const params = {
             Bucket: this._awsBucketName,
             Key: key,
             VersionId: dataStoreVersionId,
-            Range: range ? `bytes=${range[0]}-${range[1]}` : null,
-        }).on('success', response => {
-            log.trace(`${this.type} GET request response headers`,
-                { responseHeaders: response.httpResponse.headers,
-                    backendType: this.clientType });
-        });
-        const stream = request.createReadStream();
-
-        // modify the stream destroy() behavior to also abort the HTTP request
-        const streamDestroy = stream._destroy.bind(stream);
-        stream._destroy = (err, cb) => {
-            log.debug('aborting GET request in progress', { objectGetInfo });
-            request.abort();
-            if (streamDestroy) {
-                streamDestroy(err, cb);
-            }
+            Range: range ? `bytes=${range[0]}-${range[1]}` : undefined,
         };
-        stream.on('error', err => {
-            let logLevel;
-            if (err.code === 'NotFound') {
-                logLevel = 'info';
+        this._client.send(new GetObjectCommand(params)).then(data => {
+            // Always return an object with .createReadStream for test compatibility
+            const stream = data.Body;
+            const abort = () => {
+                if (isAborted) return; // Prevent multiple abort calls
+                isAborted = true;
+                
+                if (stream) {
+                    if (typeof stream.destroy === 'function') {
+                        stream.destroy();
+                    } else if (typeof stream.abort === 'function') {
+                        stream.abort();
+                    } else if (typeof stream.close === 'function') {
+                        stream.close();
+                    } else if (typeof stream.end === 'function') {
+                        stream.end();
+                    }
+                    
+                    if (typeof stream.removeAllListeners === 'function') {
+                        stream.removeAllListeners();
+                    }
+                }
+            };
+            if (stream && !stream.abort) {
+            stream.abort = abort;
+            }
+            const response = {
+                createReadStream: () => stream,
+                Body: stream,
+                abort,
+                destroy: abort,
+            };
+            return callback(null, response);
```

Comment:
```suggestion
            return callback(null, {
                createReadStream: () => stream,
                Body: stream,
                abort,
                destroy,
            });
```

(with the new `abort` variable name)